org.osgi.test.cases.cm/src/org/osgi/test/cases/cm/junit/ConfigurationPluginTests.java

```diff
@@ -250,14 +292,15 @@ public void testRankingForManagedService() throws Exception {
 		// delete configuration
 		context.invocationOrder.clear();
 		synchronizer.resetCount();
-		cm.getConfiguration(pid).delete();
+		cm.waitForService(500).getConfiguration(pid).delete();
 		trace("Wait until the ManagedService has gotten the delete");
 
 		assertTrue("Update done",
 				synchronizer.waitForSignal(SIGNAL_WAITING_TIME));
 		assertNull(synchronizer.getProps());
 		assertTrue(context.invocationOrder.isEmpty());
 		assertTrue(context.errors.isEmpty());
+		reg.unregister();
```

Comment:
This service remaining in the registry may be the root cause of the issues

Reply:
Yes.  The unregisting of the managed services is an issue. I expected that the closeable bundlecontext of osgi-tests will unregister this. Bur this did not happed. Need more investigation here